csr.py

```
@@ -483,7 +530,196 @@
+            'dn_string': ', '.join([f'{comp["field"]}={comp["value"]}' for comp in subject_components])
+        }
+
+    @staticmethod
+    def _extract_extensions(csr):
+        """Extract extensions from CSR - DEPRECATED: Use _extract_modern_extensions instead."""
+        extensions = []
+
+        try:
+            # This method is deprecated - redirect to modern implementation
+            # Convert legacy CSR object to modern format by re-parsing
+            from cryptography.hazmat.primitives import serialization
```

Ruff (0.11.9)

`cryptography.hazmat.primitives.serialization` imported but unused

Remove unused import: `cryptography.hazmat.primitives.serialization`

(F401)